### ISSUE TITLE
Request: Support associations as embeddings

### ISSUE THREAD

**Issue by devinivy** [balderdashy/sails-mongo#204]
I think it would be a very powerful feature for this adapter to be able to support associations in the form of embeddings/denormalization.  The would put the onus on write (create, update, destroy, save, add, remove) operations to keep embeddings up-to-date on both sides of *-to-many associations, virtualizing junction table schemas into instructions for embeddings.  I think it would be a large project, but it would allow Waterline to take advantage of denormalization in a NoSQL setting.  Could be very powerful!  No doubt, it would be a completely non-trivial project.

Any thoughts on this from the community?

**Comment by leobudima** [comment]
+1

**Comment by arxpoetica** [comment]
see for more info: http://blog.mongodb.org/post/87892923503/6-rules-of-thumb-for-mongodb-schema-design-part-2

**Comment by mattapperson** [comment]
+1

**Comment by elennaro** [comment]
+1

**Comment by dottodot** [comment]
This is even more important now as it sounds like they are going to remove associations from sails

https://github.com/balderdashy/sails/issues/2731#event-415488736

**Comment by sailsbot** [comment]
Marcus, don't worry, we aren't removing associations- just certain types of
deeply nested create and update queries that are particularly
accident-prone. We've experienced a lot of hard-to-debug scenarios thanks
to these types of associations, and have learned the hard way that the ends
don't justify the means.

It is still possible we will expose some kind of "maxDepth" configuration
option; but that really comes down to a discussion with @particlebanana
on what's best for the maintainability of the code base. If you'd be
interested in helping as a reviewer for that change, let me know.

On Tuesday, September 22, 2015, Marcus Williams <notifications@github.com>
wrote:

> This is even more important now as it sounds like they are going to remove
> associations from sails
>
> balderdashy/sails#2731 (comment)
> <https://github.com/balderdashy/sails/issues/2731#event-415488736>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balderdashy/sails-mongo/issues/204#issuecomment-142202076>
> .
>


-- 

Mike McNeil

CEO/founder Treeline | creator of Sails.js <http://sailsjs.org>

@mikermcneil <http://twitter.com/mikermcneil> | treeline.io


                                                         .
                                              .         ;
                 .              .              ;%     ;;
                   ,           ,                :;%  %;
                    :         ;                   :;%;'     .,
           ,.        %;     %;            ;        %;'    ,;
             ;       ;%;  %%;        ,     %;    ;%;    ,%'
              %;       %;%;      ,  ;       %;  ;%;   ,%;'
               ;%;      %;        ;%;        % ;%;  ,%;'
                `%;.     ;%;     %;'         `;%%;.%;'
                 `:;%.    ;%%. %@;        %; ;@%;%'
                    `:%;.  :;bd%;          %;@%;'
                      `@%:.  :;%.         ;@@%;'
                        `@%.  `;@%.      ;@@%;
                          `@%%. `@%%    ;@@%;
                            ;@%. :@%%  %@@%;
                              %@bd%%%bd%%:;
                                #@%%%%%:;;
                                %@@%%%::;
                                %@@@%(o);  . '
                                %@@@o%;:(.,'
                            `.. %@@@o%::;
                               `)@@@o%::;
                                %@@(o)::;
                               .%@@@@%::;
                               ;%@@@@%::;.
                              ;%@@@@%%:;;;.
                          ...;%@@@@@%%:;;;;,..

**Comment by harianus** [comment]
Isn't this the same as https://github.com/balderdashy/sails-mongo/issues/44? The embedded documents / associations feature in Sails.js is the only thing what is stopping me to use this framework for my new projects. It is somewhere on the feature list, but is there also an ETA, or can the community do something to push this feature forward?

**Comment by martolini** [comment]
+1, if sails with mongoose were out of the box, sails would to me be a nobrainer to use. The core NoSQL features are essential.

**Comment by sailsbot** [comment]
Thanks for posting, @devinivy.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by elennaro** [comment]
This bot is killing us :(

**Comment by devinivy** [comment]
Should probably go into a roadmap.md.

**Comment by sailsbot** [comment]
Thanks for posting, @devinivy.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by harianus** [comment]
This one is for @sailsbot:

<img src="https://cloud.githubusercontent.com/assets/1079135/12353142/51e83124-bb8c-11e5-8b2a-7a943550778f.gif" width="80%">
